hibernate-core/src/main/java/org/hibernate/type/OneToOneType.java

```diff
@@ -203,36 +196,25 @@ public boolean useLHSPrimaryKey() {
 
 	@Override
 	public Serializable disassemble(Object value, SharedSessionContractImplementor session, Object owner) throws HibernateException {
-		if (value == null) {
-			return null;
-		}
-
-		Object id = ForeignKeys.getEntityIdentifierIfNotUnsaved( getAssociatedEntityName(), value, session );
-
-		if ( id == null ) {
-			throw new AssertionFailure(
-				"cannot cache a reference to an object with a null id: " +
-				getAssociatedEntityName()
-			);
-		}
-
-		return getIdentifierType( session ).disassemble( id, session, owner );
+		return null;
 	}
 
 	@Override
 	public Object assemble(Serializable oid, SharedSessionContractImplementor session, Object owner) throws HibernateException {
-		//the owner of the association is not the owner of the id
-		Serializable id = ( Serializable ) getIdentifierType( session ).assemble( oid, session, null );
-
-		if ( id == null ) {
-			return null;
-		}
-
-		return resolveIdentifier( id, session );
+		//this should be a call to resolve(), not resolveIdentifier(),
+		//'cos it might be a property-ref, and we did not cache the
```

Comment:
"'cos" ?

Reply:
I think it stands for because

Reply:
I know, but let's try to avoid getting too informal?

Reply:
I really didn't even noticed it, just reverted the change, anyway changed it